app/email/email.go

```diff
@@ -480,43 +413,41 @@ func (es *EmailService) sendMfaChangeEmail(email string, activated bool, locale,
 	}
 	data.Props["Warning"] = T("api.templates.email_warning")
 
-	body, err := es.srv.TemplatesContainer().RenderToString("mfa_change_body", data)
+	body, err := es.templatesContainer.RenderToString("mfa_change_body", data)
 	if err != nil {
-		return model.NewAppError("SendMfaChangeEmail", "api.user.send_mfa_change_email.error", nil, err.Error(), http.StatusInternalServerError)
+		return err
 	}
 
 	if err := es.sendMail(email, subject, body); err != nil {
-		return model.NewAppError("SendMfaChangeEmail", "api.user.send_mfa_change_email.error", nil, err.Error(), http.StatusInternalServerError)
+		return err
 	}
 
 	return nil
 }
 
-func (es *EmailService) SendInviteEmails(team *model.Team, senderName string, senderUserId string, invites []string, siteURL string) *model.AppError {
+func (es *Service) SendInviteEmails(team *model.Team, senderName string, senderUserId string, invites []string, siteURL string) error {
 	if es.PerHourEmailRateLimiter == nil {
-		return model.NewAppError("SendInviteEmails", "app.email.no_rate_limiter.app_error", nil, fmt.Sprintf("user_id=%s, team_id=%s", senderUserId, team.Id), http.StatusInternalServerError)
+		return NoRateLimiterError
 	}
 	rateLimited, result, err := es.PerHourEmailRateLimiter.RateLimit(senderUserId, len(invites))
 	if err != nil {
-		return model.NewAppError("SendInviteEmails", "app.email.setup_rate_limiter.app_error", nil, fmt.Sprintf("user_id=%s, team_id=%s, error=%v", senderUserId, team.Id, err), http.StatusInternalServerError)
+		return SetupRateLimiterError
 	}
 
 	if rateLimited {
-		return model.NewAppError("SendInviteEmails",
-			"app.email.rate_limit_exceeded.app_error", map[string]interface{}{"RetryAfter": result.RetryAfter.String(), "ResetAfter": result.ResetAfter.String()},
-			fmt.Sprintf("user_id=%s, team_id=%s, retry_after_secs=%f, reset_after_secs=%f",
-				senderUserId, team.Id, result.RetryAfter.Seconds(), result.ResetAfter.Seconds()),
-			http.StatusRequestEntityTooLarge)
+		mlog.Error("rate limit exceeded", mlog.Duration("RetryAfter", result.RetryAfter), mlog.Duration("ResetAfter", result.ResetAfter), mlog.String("user_id", senderUserId),
```

Comment:
Any reason we are logging and returning the error? Can we not log it at the call-site?

Reply:
This function is called several times and TBH I was not comfortable doing so but I think it may be simpler to keep this way as it would need to be changed for all occurrences if it ever changes.

Reply:
ok